### PR TITLE
fix: use index cache in artifact commands

### PR DESCRIPTION
**Pull request by loresuso**
Signed-off-by: Lorenzo Susini <susinilorenzo1@gmail.com>

**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

/kind bug

> /kind cleanup

> /kind design

> /kind documentation

> /kind failing-test

> /kind feature

> /kind flaky-test

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

> /area library

/area cli

> /area tests

> /area examples

**What this PR does / why we need it**:
Also `artifact search` and `artifact list` commands need to use the configured indexes to provide their output.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: cmd/artifact/info/info.go [hunk 1]
```diff
@@ -59,22 +59,18 @@ func NewArtifactInfoCmd(ctx context.Context, opt *options.CommonOptions) *cobra.
...
+	indexCache, err := cache.NewFromConfig(ctx, config.IndexesFile, config.IndexesDir, indexes)
+	o.Printer.CheckErr(err)
 
 	var data [][]string
 	for _, name := range args {
 		var ref string
 		parsedRef, err := registry.ParseReference(name)
 		if err != nil {
-			entry, ok := mergedIndexes.EntryByName(name)
+			entry, ok := indexCache.MergedIndexes.EntryByName(name)
```
Comment (alacuku) [inline cmd/artifact/info/info.go:73]:
```suggestion
			entry, ok := o.IndexCache.EntryByName(name)
```

File: cmd/artifact/search/artifact_search.go [hunk 2]
```diff
@@ -75,24 +74,20 @@ func NewArtifactSearchCmd(ctx context.Context, opt *options.CommonOptions) *cobr
...
-		return err
-	}
+	indexes, err := config.Indexes()
+	o.Printer.CheckErr(err)
 
-	mergedIndexes, err := utils.Indexes(indexConfig, config.IndexesDir)
-	if err != nil {
-		return err
-	}
+	indexCache, err := cache.NewFromConfig(ctx, config.IndexesFile, config.IndexesDir, indexes)
```
Comment (alacuku) [inline cmd/artifact/search/artifact_search.go:80]: Ditto.

File: cmd/artifact/search/artifact_search.go [hunk 3]
```diff
@@ -75,24 +74,20 @@ func NewArtifactSearchCmd(ctx context.Context, opt *options.CommonOptions) *cobr
...
+	o.Printer.CheckErr(err)
 
-	resultEntries := mergedIndexes.SearchByKeywords(o.minScore, args...)
+	resultEntries := indexCache.MergedIndexes.SearchByKeywords(o.minScore, args...)
```
Comment (alacuku) [inline cmd/artifact/search/artifact_search.go:83]:
```suggestion
	resultEntries := o.IndexCache.SearchByKeywords(o.minScore, args...)
```

File: cmd/artifact/search/artifact_search.go [hunk 4]
```diff
@@ -75,24 +74,20 @@ func NewArtifactSearchCmd(ctx context.Context, opt *options.CommonOptions) *cobr
...
 
 	var data [][]string
 	for _, entry := range resultEntries {
 		if o.artifactType != "" && o.artifactType != oci.ArtifactType(entry.Type) {
 			continue
 		}
-		indexName := mergedIndexes.IndexByEntry(entry).Name
+		indexName := indexCache.MergedIndexes.IndexByEntry(entry).Name
```
Comment (alacuku) [inline cmd/artifact/search/artifact_search.go:90]:
```suggestion
		indexName := o.IndexCache.IndexByEntry(entry).Name
```

**Review by alacuku — CHANGES_REQUESTED** [review]

**Comment by loresuso** [comment]
You are right @alacuku ! This will become even much easier!

**Review by alacuku — APPROVED** [review]

**Review by leogr — APPROVED** [review]
nice :+1:

**Pull request merged**